streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableAggregate.java

```diff
@@ -104,13 +106,16 @@ public void process(final K key, final Change<V> value) {
                 }
 
                 newAgg = add.apply(key, value.newValue, initializedAgg);
+                if (oldAggAndTimestamp != null) {
```

Comment:
I found this logic is a bit hard to read, eventually it means `if (oldValue != null || newValue != null) we always check if oldAggAndTimestamp should be updated`.

Could we just extract this part independently for reading purposes?

Reply:
Not sure if I can follow. Can you elaborate?